Meta/ladybird.py

```diff
@@ -0,0 +1,295 @@
+    user_vars_cmake_module.parent.mkdir(parents=True, exist_ok=True)
+
+    with open(user_vars_cmake_module, 'w') as f:
+        f.writelines([
+            f'set(PKGCONFIG {pkg_config})',
+            f'set(GN {gn})',
+            '',
+        ])
+
+    return cmake_args
+
+
+def _configure_build_env(**kwargs):
+    preset = kwargs.get('preset')
+    cc = kwargs.get('cc')
+    cxx = kwargs.get('cxx')
+    cmake_args = []
+    cmake_args.extend([
+        f'-DCMAKE_C_COMPILER={cc}',
+        f'-DCMAKE_CXX_COMPILER={cxx}',
+    ])
+    _ensure_ladybird_source_dir()
+    lb_source_dir = Path(os.environ.get('LADYBIRD_SOURCE_DIR'))
+
+    build_root_dir = lb_source_dir / 'Build'
+
+    known_presets = {
+        'default': build_root_dir / 'release',
+        'windows_ci_ninja': build_root_dir / 'release',
+        'windows_dev_ninja': build_root_dir / 'debug',
+        'windows_dev_msbuild': build_root_dir / 'debug',
+        'Debug': build_root_dir / 'debug',
+        'Sanitizer': build_root_dir / 'sanitizers',
+        'Distribution': build_root_dir / 'distribution',
+    }
+    if preset not in known_presets:
+        print(f'Unknown build preset "{preset}"', file=sys.stderr)
+        sys.exit(1)
+
+    build_preset_dir = known_presets.get(preset)
+    vcpkg_root = str(build_root_dir / 'vcpkg')
+    os.environ['VCPKG_ROOT'] = vcpkg_root
+    os.environ['PATH'] += os.pathsep + str(lb_source_dir.joinpath('Toolchain', 'Local', 'cmake', 'bin'))
+    os.environ['PATH'] += os.pathsep + str(vcpkg_root)
+    return lb_source_dir, build_preset_dir, cmake_args
+
+
+def _build_vcpkg():
+    sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'Toolchain'))
+    from BuildVcpkg import main as build_vcpkg
+    build_vcpkg()
+
+
+def _validate_cmake_version():
+    import re
+    cmake_pls_install_msg = 'Please install CMake version 3.25 or newer.'
+
+    try:
+        cmake_version_output = subprocess.check_output(
+            [
+                'cmake',
+                '--version',
+            ],
+            text=True
+        ).strip()
+
+        version_match = re.search(r'version\s+(\d+)\.(\d+)\.(\d+)?', cmake_version_output)
+        if version_match:
+            major = int(version_match.group(1))
+            minor = int(version_match.group(2))
+            patch = int(version_match.group(3))
+            if major < 3 or (major == 3 and minor < 25):
+                print(f'CMake version {major}.{minor}.{patch} is too old. {cmake_pls_install_msg}', file=sys.stderr)
+                sys.exit(1)
+        else:
+            print(f'Unable to determine CMake version. {cmake_pls_install_msg}', file=sys.stderr)
+            sys.exit(1)
+    except subprocess.CalledProcessError as e:
+        _print_process_stderr(e, f'CMake not found. {cmake_pls_install_msg}\n')
+        sys.exit(1)
+
+
+def _ensure_ladybird_source_dir():
+    ladybird_source_dir = os.environ.get('LADYBIRD_SOURCE_DIR')
```

Comment:
why not wrap this in Path here, and have it be a pathlib.Path for the rest of the method? below you convert from path to string and then just return it as path at the end of the method.